### ISSUE TITLE
Ensure GraphQL API and schemas supports `List` types

### ISSUE THREAD

**Issue by deekerno** [FuelLabs/fuel-indexer#638]
When #503 is completed, we will be able to decode `Vec` types from Sway contracts. However, we still need to add the ability to store the vectors in the database and retrieve them in response to GraphQL API queries.

**Comment by ra0x3** [comment]
@deekerno 
- Thanks for opening this
- I would clarify that querying GraphQL list types doesn't require storing any type of vector/list/array
- With GraphQL syntax `[Foo]` just means an array of `Foo` items. (e.g., `SELECT * from foo`)
- It doesn't necessarily mean that `Foo` is stored as a vector
- I also think we should discuss (maybe in a team meeting) how much GraphQL functionality we want to support cc @rvmelkonian 
- Foreign keys are must-haves, list types (when querying) are must-haves, filtering (a bit) is obvious, we don't use mutations at all yet. But past that, I feel that if we don't draw a hard line at what we do/don't support, we might walk into a little quagmire since we don't have the resources to support much more than the GraphQL basics

**Comment by deekerno** [comment]
Whoops, yeah, "storing the vectors in the database" was a bit of a flub on my part as I envisioned us abstracting that away from the user anyways. 😅 but I agree, we should absolutely talk about how much we're willing to implement and then only do that (barring significant requests from the community).

**Comment by Voxelot** [comment]
is this a dupe of #888 ?

**Comment by deekerno** [comment]
Yeah, I would consider it a dupe as #888 has more information in it now. I'll close this to centralize any discussion regarding list types on the other issue.
